### ISSUE TITLE
Add `concat()` test using `ArrayProxy`

### ISSUE THREAD

**Issue by Turbo87** [kellyselden/ember-awesome-macros#485]
The `Ember.Array` interface does not define a `concat()` method, but the `concat()` macro seems to use it as if it exists. We might have an issue here and we should at least add a test to make sure whether it works or not.

**Comment by kellyselden** [comment]
Agreed. Anyone watching is free to help. @Turbo87 Can now help review and merge.

**Comment by baljeet03** [comment]
@Turbo87 I would like to do that. I had a little doubt regarding the test case though. Should I create a `Ember.ArrayProxy` object and convert it to `array` and test it or something else.
